motionplan/plan.go

```diff
@@ -0,0 +1,154 @@
+package motionplan
+
+import (
+	"errors"
+	"fmt"
+	"math"
+
+	pb "go.viam.com/api/service/motion/v1"
+	"go.viam.com/rdk/motionplan/ik"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/spatialmath"
+)
+
+type Plan struct {
+	Trajectory
+	Path
+
+	// nodes corresponding to inputs can be cached with the Plan for easy conversion back into a form usable by RRT
+	// depending on how the Trajectory is constructed these may be nil and should be computed before usage
+	nodes []node
+}
+
+func newPlan(solution []node, sf *solverFrame, relative bool) (*Plan, error) {
+	traj := sf.nodesToTrajectory(solution)
+	path, err := newPath(solution, sf)
+	if err != nil {
+		return nil, err
+	}
+	return &Plan{
+		Trajectory: traj,
+		Path:       path,
+		nodes:      solution,
+	}, nil
+}
+
+// TODO: can probably think of a better name for this function
+func (plan *Plan) Remaining(waypointIndex int) *Plan {
+	// TODO: I don't think a deep copy should be necessary here but maybe?
+	return &Plan{
```

Comment:
> How will a caller know what index the plan is on?

I expect that this would be tracked by `waypointIndex` of the `moveRequest` struct.
Do you have concerns about this construction?

Reply:
Addressed 1 and 3 I think.  2 is tracked in a different comment.  Going to resolve this thread.